src/app/pages/AccountPage/index.tsx

```diff
@@ -135,6 +136,14 @@ export function AccountPage(props: Props) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [dispatch, walletsPublicKeys.join(','), selectedNetwork])
 
+  if (!isValidAddress(address!)) {
```

Comment:
Can this be checked at the router level?

Reply:
I'm not aware of any method for doing conditional route matching with React Router _based on the current value of dynamic segments_. (Although I'm not too familiar with React Router; I usually prefer to use my own router implementation.) As far as I can tell, it's not supported.

Reply:
awful :/

I think we should add a wrapper around AccountPage to validate then, so AccountPage doesn't dispatch a bunch of actions we need to discard

Reply:
Yes, I was thinking about the same thing today.

Reply:
>  we should add a wrapper around AccountPage to validate then, so AccountPage doesn't dispatch a bunch of actions we need to discard

OK I have added the wrapper.